server/container_remove.go

```diff
@@ -23,7 +23,7 @@ func (s *Server) RemoveContainer(ctx context.Context, req *pb.RemoveContainerReq
 	}
 
 	cState := s.runtime.ContainerStatus(c)
-	if cState.Status == oci.ContainerStateCreated || cState.Status == oci.ContainerStateRunning {
+	if cState.Status == oci.ContainerStateRunning {
```

Comment:
@mrunalp What is the difference between a process started and exec'ed ?
I think it would be clear to have something like this:
```
Created ==> the pod or the container is ready and this can happen after a container create. But nothing has been started, meaning that no workload is running.
Running ==> We started the container with container start, and we have the initial workload running. We can execute additional processes on the same container with container exec.
Stopped ==> A running container has been stopped by container stop.
container create can only be called if no other container with the same name already exist.
container start can only be called on a Created or Stopped container.
container stop can only be called on a Running container.
container remove can only be called on a Stopped or Created container
```

Does that make sense ?